src/main/java/com/github/tomakehurst/wiremock/http/ssl/CertificateAuthority.java

```diff
@@ -37,9 +38,7 @@ public class CertificateAuthority {
 
   public CertificateAuthority(X509Certificate[] certificateChain, PrivateKey key) {
     this.certificateChain = requireNonNull(certificateChain);
-    if (certificateChain.length == 0) {
-      throw new IllegalArgumentException("Chain must have entries");
-    }
+    checkState(certificateChain.length == 0, "Chain must have entries");
```

Comment:
Hi, it looks like this would introduce a change in behaviour.  The original code throws an `IllegalArgumentException` where as the `checkState` function throws an `IllegalStateException`.  I think it would be better to maintain the original exception and perhaps use the `checkParameter` method instead

Reply:
I have changed this